mlir/lib/Dialect/Rock/Transforms/GridwiseGemmToBlockwise.cpp

```diff
@@ -127,8 +127,7 @@ computeCopyPerThread(Type elementType, int64_t copyPerThread, int64_t kPerBlock,
     copyDPerThread = math_util::gcd(maxVlen, copyPerThread);
     copyKPerThread = copyPerThread / copyDPerThread;
   } else {
-    copyKPerThread =
-        math_util::gcd(maxVlen, math_util::gcd(kpack, copyPerThread));
+    copyKPerThread = math_util::gcd(maxVlen, copyPerThread);
```

Comment:
`copyPerThread = (KPerBlock * MPerBlock) / blockSize`

where `KPerBlock = kPack * kPacksPerBlock`

and `blockSize = ((MPerBlock * NPerBlock) / (MPerWave * NPerWave)) * waveSize` 

therefore 
`copyPerThread =  ((KPerBlock  * (MPerWave * NPerWave)  * WaveSize) / NPerBlock)`

So it seem GCD should always be kPack
`gcd(kPack, copyPerThread) =  kPack`

Reply:
So previous logic was indeed limiting `copyKPerThread` to `kPack`

Reply:
I think we should run some more experiments on larger bitwidth dtypes. I am not sure how it affects the register pressures. 

You can change rocMLIR SHA similar to this PR on MIGraphX side https://github.com/ROCm/AMDMIGraphX/pull/3743 

and MIGraphX CI would run more models with this change.